src/backend/renderer/element/mod.rs

```diff
@@ -89,6 +96,50 @@ pub enum UnderlyingStorage<'a, R: Renderer> {
     External(&'a R::TextureId),
 }
 
+/// Defines the possible states of an element
+/// after rendering
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum RenderElementState {
+    /// The element was rendered
+    Rendered,
+    /// The element was directly scanned out
+    ScannedOut,
+    /// The element was skipped
+    Skipped,
+}
+
+/// Report for an output render operation
+#[derive(Debug, Clone)]
+pub struct OutputRenderReport {
+    /// Holds the damage for presenting the bound buffer
+    pub damage: Option<Vec<Rectangle<i32, Physical>>>,
```

Comment:
I'd drop fallability and just make the empty vec be equivalent to None.

Reply:
Right, we could also check for an empty vec